rsocket-core/src/main/java/io/rsocket/exceptions/RSocketException.java

```diff
@@ -16,41 +16,47 @@
 
 package io.rsocket.exceptions;
 
-import java.util.Objects;
+import io.rsocket.RSocketErrorException;
 import reactor.util.annotation.Nullable;
 
-/** The root of the RSocket exception hierarchy. */
-public abstract class RSocketException extends RuntimeException {
+/**
+ * The root of the RSocket exception hierarchy.
+ *
+ * @deprecated please use {@link RSocketErrorException} instead
+ */
+@Deprecated
+public abstract class RSocketException extends RSocketErrorException {
 
   private static final long serialVersionUID = 2912815394105575423L;
 
   /**
-   * Constructs a new exception with the specified message.
+   * Constructs a new exception with the specified message and error code 0x201 (Application error).
    *
    * @param message the message
-   * @throws NullPointerException if {@code message} is {@code null}
    */
   public RSocketException(String message) {
-    super(Objects.requireNonNull(message, "message must not be null"));
+    this(message, null);
   }
 
   /**
-   * Constructs a new exception with the specified message and cause.
+   * Constructs a new exception with the specified message and cause and error code 0x201
+   * (Application error).
    *
    * @param message the message
    * @param cause the cause of this exception
-   * @throws NullPointerException if {@code message} is {@code null}
    */
   public RSocketException(String message, @Nullable Throwable cause) {
-    super(Objects.requireNonNull(message, "message must not be null"), cause);
+    super(0x00000201, message, cause);
```

Comment:
can we use `ErrorType.APPLICATION_ERROR` here. In any case, we have ref to `ErrorType`
```suggestion
    super(ErrorType.APPLICATION_ERROR, message, cause);
```

Reply:
Yes, indeed, this should be a constant. I've suggested a further change to use the constants from `ErrorFrameFlyweight`.